packages/eventual-send/src/index.js

```diff
@@ -54,15 +54,46 @@ export function makeHandledPromise(Promise) {
   let presenceToPromise;
   let promiseToHandler;
   let promiseToPresence; // only for HandledPromise.unwrap
+  let promiseToForwardedPromise; // forwarding, union-find-ish
```

Comment:
Confusing name. The key is the forwarded promise. The value is potentially any promise (any handled promise?). Thus, a more accurate name would be `forwardedPromiseToPromise` or `promiseToTargetPromise`. I prefer the first to avoid needing to explain what we mean by "target" here.